controllers/cluster_controller.go

```diff
@@ -544,13 +553,38 @@ func aggregatedGeneration(config *cluster.Config) int64 {
 }
 
 func (r *ClusterReconciler) setBundlesRef(ctx context.Context, clus *anywherev1.Cluster) error {
+	mgmtCluster, err := validateManagementClusterExists(ctx, clus, r.client)
+	if err != nil {
+		return err
+	}
+
+	clus.Spec.BundlesRef = mgmtCluster.Spec.BundlesRef
+	return nil
+}
+
+func (r *ClusterReconciler) setDefaultEksaVersion(ctx context.Context, clus *anywherev1.Cluster) error {
+	mgmtCluster, err := validateManagementClusterExists(ctx, clus, r.client)
+	if err != nil {
+		return err
+	}
+
+	if mgmtCluster.Spec.EksaVersion == nil {
+		clus.Status.FailureMessage = ptr.String("Management cluster cannot have nil EksaVersion")
+		return fmt.Errorf("mgmt cluster cannot have nil EksaVersion")
+	}
+
+	clus.Spec.EksaVersion = mgmtCluster.Spec.EksaVersion
+	return nil
+}
+
+func validateManagementClusterExists(ctx context.Context, clus *anywherev1.Cluster, client client.Client) (*anywherev1.Cluster, error) {
```

Comment:
why is a validate method returning an object?
This signature is a bit confusing

I think we should clean this up a bit

Reply:
i fixed the method's name